10-block/2/exercise4_2.js

```diff
@@ -0,0 +1,34 @@
+const messageDelay = () => Math.floor(Math.random() * 5000);
+
+const getMarsTemperature = () => {
+  const maxTemperature = 58;
+  return Math.floor(Math.random() * maxTemperature);
+};
+
+const toFahrenheit = degreeCelsius => (degreeCelsius * 9) / 5 + 32;
+const temperatureInFahrenheit = temperature =>
+  console.log(`It is currently ${toFahrenheit(temperature)}ºF at Mars`);
+const greet = temperature =>
+  console.log(
+    `Hi there! Curiosity here. Right now is ${temperature}ºC at Mars`
+  );
+
+const handleError = errorReason =>
+  console.log(`Error getting temperature: ${errorReason}`);
+
+const isRobotBusy = () => (Math.floor(Math.random() * 101) < 60 ? false : true);
+
+// definição da função sendMarsTemperature...
+const sendMarsTemperature = callback => {
+  setTimeout(() => {
+    isRobotBusy()
+      ? handleError('Robot is busy')
+      : callback(getMarsTemperature());
+  }, 1000);
+};
```

Comment:
Boa Panta, uma outra opção usando cois callbacks:
```suggestion
const sendMarsTemperature = (callback, callbackError) => {
  setTimeout(() => {
    const lucky = Math.random() >= 0.4;
    if (lucky) return callback(getMarsTemperature());
    return callbackError('Robot is busy');
  }, messageDelay());
};
```

Reply:
Boa!! Tinha errado mesmo nesse, a ideia realmente é usar dois callbacks!!
Valeu Ale :fist_oncoming: